lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -1606,6 +1826,50 @@ impl ServerDeviceAnnounceResponse {
     }
 }
 
+#[derive(Debug, Clone, ToPrimitive)]
+#[repr(u32)]
+#[allow(non_camel_case_types)]
+#[allow(dead_code)]
+enum ClientNameRequestUnicodeFlag {
+    Ascii = 0x0,
+    Unicode = 0x1,
+}
+
+/// 2.2.2.4 Client Name Request (DR_CORE_CLIENT_NAME_REQ)
+/// https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/902497f1-3b1c-4aee-95f8-1668f9b7b7d2
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
+pub struct ClientNameRequest {
+    unicode_flag: ClientNameRequestUnicodeFlag,
+    computer_name: String,
+}
+
+impl ClientNameRequest {
+    fn new(unicode_flag: ClientNameRequestUnicodeFlag, computer_name: String) -> Self {
+        Self {
+            unicode_flag,
+            computer_name,
+        }
+    }
+
+    fn encode(&self) -> RdpResult<Vec<u8>> {
+        let mut w = vec![];
+        w.write_u32::<LittleEndian>(self.unicode_flag.clone() as u32)?;
+        // CodePage (4 bytes): A 32-bit unsigned integer that specifies the code page of the ComputerName field; it MUST be set to 0.
+        w.write_u32::<LittleEndian>(0x0)?;
+
+        let computer_name_data = match self.unicode_flag {
+            ClientNameRequestUnicodeFlag::Ascii => self.computer_name.as_bytes().to_vec(),
+            ClientNameRequestUnicodeFlag::Unicode => util::to_utf8(&self.computer_name),
```

Comment:
Are you sure this works? Usually when Windows says "unicode" they mean UTF16-LE.

Reply:
Nice catch! I've changed it to the other unicode function from the same util module that will handle `UTF16-LE` unicode